### PR TITLE
perf: don't update on scroll/rect changes unless virtualized range changes

### PR DESCRIPTION
**Pull request by mogelbrod**
Fixes https://github.com/tannerlinsley/react-virtual/issues/46

- Avoid state updates caused by `useScroll()` that doesn't cause the virtualized range to change
- Remove the unnecessary `reversedMeasurements` array and `scrollOffsetPlusOuterSize` variable
- Remove the no longer used `useScroll` hook

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -143,6 +104,36 @@ export function useVirtual({
...
+    element.addEventListener('scroll', handler, {
+      capture: false,
+      passive: true,
+    })
+
+    return () => {
+      element.removeEventListener('scroll', handler)
+    }
+  }, [parentRef, scrollKey])
+  //*/
```
Comment (mogelbrod) [inline src/index.js:135]: For some reason I couldn't get the hook to work with the existing `useScroll()`. Sometimes the `setRange()` callback would only be called on the initial mount and not during scrolling. Any ideas on how to fix this?

```suggestion
  useScroll(parentRef, ({ [scrollKey]: scrollOffset }) => {
    latestRef.current.scrollOffset = scrollOffset
    setRange(prevRange => {
      console.log('setRange') // FIXME: Sometimes never triggers after the initial mount
      return calculateRange(latestRef.current, prevRange)
    })
  })
```

**Comment by mogelbrod** [comment]
> Looks good! Can you rebase with the last 2 PRs that came in?

Done!

I also removed the `useScroll` hook that [I couldn't get to work](https://github.com/tannerlinsley/react-virtual/pull/48#discussion_r470160422) in favour of some (hopefully) more streamlined code.

File: src/index.js [hunk 2]
```diff
@@ -52,69 +43,56 @@ export function useVirtual({
...
-  }
+    element.addEventListener('scroll', onScroll, {
+      capture: false,
+      passive: true,
+    })
+
+    return () => {
+      element.removeEventListener('scroll', onScroll)
+    }
+  }, [parentRef, scrollKey, size /* required */])
```
Comment (quanbrew) [inline src/index.js:90]: I tried this PR, list doesn't corrently rendering. fixed by 
```suggestion
  }, [parentRef.current, scrollKey, size /* required */])
```

Comment (mogelbrod) [reply src/index.js:90]: Good catch! The old `useScroll` hook had some code that handled the parentRef changing which I had omitted. Looks like this simple change should solve that 👍

**Comment by mogelbrod** [comment]
@tannerlinsley This PR should be ready for another pass now 👍

**Comment by tannerlinsley** [comment]
Looks good!

**Comment by gausie** [comment]
@mogelbrod @tannerlinsley I think this change caused https://github.com/tannerlinsley/react-virtual/issues/68 and https://github.com/tannerlinsley/react-virtual/issues/56 because it attempts to interact with the parentRef element without checking whether it has been assigned (e.g. https://github.com/tannerlinsley/react-virtual/blob/master/src/index.js#L83)

**Pull request merged**